crates/aptos/src/common/types.rs

```diff
@@ -560,3 +534,81 @@ pub fn load_account_arg(str: &str) -> Result<AccountAddress, CliError> {
         ))
     }
 }
+
+/// A common trait for all CLI commands to have consistent outputs
+#[async_trait]
+pub trait CliCommand<T: Serialize + Send>: Sized + Send {
+    /// Returns a name for logging purposes
+    fn command_name(&self) -> &'static str;
+
+    /// Executes the command, returning a command specific type
+    async fn execute(self) -> CliTypedResult<T>;
+
+    /// Executes the command, and serializes it to the common JSON output type
+    async fn execute_serialized(self) -> CliResult {
+        let command_name = self.command_name();
+        to_common_result(command_name, self.execute().await).await
+    }
+
+    /// Executes the command, and throws away Ok(result) for the string Success
+    async fn execute_serialized_success(self) -> CliResult {
+        let command_name = self.command_name();
+        to_common_success_result(command_name, self.execute().await).await
+    }
+}
+
+/// A shorted transaction output
+#[derive(Clone, Debug, Default, Serialize)]
+pub struct TransactionSummary {
+    changes: BTreeMap<String, serde_json::Value>,
+    gas_used: Option<u64>,
+    success: bool,
+    version: Option<u64>,
+    vm_status: String,
+}
+
+impl From<Transaction> for TransactionSummary {
+    fn from(transaction: Transaction) -> Self {
+        let mut summary = TransactionSummary {
+            success: transaction.success(),
+            version: transaction.version(),
+            vm_status: transaction.vm_status(),
+            ..Default::default()
+        };
+
+        if let Ok(info) = transaction.transaction_info() {
+            summary.gas_used = Some(info.gas_used.0);
+            summary.changes = info
+                .changes
+                .iter()
+                .map(|change| match change {
+                    WriteSetChange::DeleteModule { module, .. } => {
```

Comment:
In a future world it would be nice if "WriteSetChange" had consistent formatting, so regardless of where it is being displayed, we'll see the same output,. e.g., there is already a `type_str` name for the modules.

Reply:
Yeah, the problem here is it has a bunch of junk around bytecode and things that really don't matter, so I was cutting out a lot of it.  We can look into that later, but at least now it doesn't have a huge output of bytes.

Reply:
Fixed, now it's a struct that serializes nicer